### ISSUE TITLE
smtpAppender error unhandled when no network present

### ISSUE THREAD

**Issue by smartaz** [nomiddlename/log4js-node#333]
Is it expected for the log4js to quit the process with the following:

log4js.smtpAppender - Error happened { [Error: getaddrinfo ENOTFOUND smtp.gmail.com]
  code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'smtp.gmail.com' }

**Comment by nomiddlename** [comment]
Related to log4js-node/log4js-node#575 - needs a standard way of handling appender errors

**Comment by pharapiak** [comment]
I have a similar situation with the TCP appender.  I'm using it as an optional appender, in parallel with console, for developers working on a multi-server application. They might be running it, or maybe not.  Or they may change the log4js configuration on the tcp-server, which would trigger a restart, and temporarily broken connections.

It would be nice to have an option to let the logging apps keep on running in the face of TCP connection failures.  

I was thinking along the lines of a callback that was passed the appender, which could decide whether to ignore the error, throw it, or maybe retry the write operation, or disable the appender.

**Comment by lamweili** [comment]
@pharapiak 

* tcpAppender (#1089) - fixed in `log4js@6.4.0`

**Comment by lamweili** [comment]
The original issue of the stmpAppender might need to be transferred to https://github.com/log4js-node/smtp.
@nomiddlename You reckon?
